Replace StringBuffer with StringBuilder inside the method

wxisme opened the pull request:
## Replace StringBuffer with StringBuilder inside the method

Using `StringBuilder` is more efficient than using `StringBuffer` without thinking about thread safety.  Therefore this PR replace StringBuffer with StringBuilder inside the method.

- [ ] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

xiangwangcheng commented:
THANKS for your contribution. For most of the cases, we don't need a thread-safe String concat.
But  could u please change your destination branch to develop?

wxisme commented:
@xiangwangcheng Yes

wxisme commented:
@xiangwangcheng I rebased develop to my branch and changed the destination branch to develop in a new PR https://github.com/apache/rocketmq/pull/1364.
